### PR TITLE
Make "see issue" consistent

### LINKED ISSUE #62976: Make references to GitHub issues in diagnostics consistent
Some diagnostics include a link to GitHub issues. From the ui testsuite, I found at least these two patterns:

```
 = note: for more information, see https://github.com/rust-lang/rust/issues/51999
```

```
 = note: for more information, see issue #36887 <https://github.com/rust-lang/rust/issues/36887>
```

These should use the same format. My preference is for the latter:

> `see issue #XXXXX <https://github.com/rust-lang/rust/issues/XXXXX>`

/cc @varkor

### PR DESCRIPTION
**Pull request by eupn**
Fixes #62976. h/t to @punitkoura for initial patterns search.

r? @varkor

### CODE REVIEW

**Review by ghost — CHANGES_REQUESTED** [review]
there are some more inconsistencies that should be fixed (at least some of them): 
>See issue #49146 \<https://github.com/rust-lang/rust/issues/49146> \\
**for more information**.", kind),

>**for more information**, see issue \\
 #57563 \<https://github.com/rust-lang/rust/issues/57563>");

>see issue #43162 \<https://github.com/rust-lang/rust/issues/43162#issuecomment-320764675>"

>**please** see issue #44136 \<https://github.com/rust-lang/rust/issues/44136>

>**for more context**, see issue #60210 \<https://github.com/rust-lang/rust/issues/60210>

>see issue #46062 \<https://github.com/rust-lang/rust/issues/46062> **for more details**

I think, the order should be consistent, as well as the inclusion of "for more information/details"

**Comment by eupn** [comment]
@BO41
> there are some more inconsistencies that should be fixed (at least some of them):
> 
> > See issue #49146 <[https://github.com/rust-lang/rust/issues/49146>](https://github.com/rust-lang/rust/issues/49146%3E) \
> > **for more information**.", kind),
> 
> > **for more information**, see issue \
> > #57563 <[https://github.com/rust-lang/rust/issues/57563>](https://github.com/rust-lang/rust/issues/57563%3E)");
> 
> > see issue #43162 <[https://github.com/rust-lang/rust/issues/43162#issuecomment-320764675>](https://github.com/rust-lang/rust/issues/43162#issuecomment-320764675%3E)"
> 
> > **please** see issue #44136 <[https://github.com/rust-lang/rust/issues/44136>](https://github.com/rust-lang/rust/issues/44136%3E)
> 
> > **for more context**, see issue #60210 <[https://github.com/rust-lang/rust/issues/60210>](https://github.com/rust-lang/rust/issues/60210%3E)
> 
> > see issue #46062 <[https://github.com/rust-lang/rust/issues/46062>](https://github.com/rust-lang/rust/issues/46062%3E) **for more details**
> 
> I think, the order should be consistent, as well as the inclusion of "for more information/details"

I should've made this consistent as well. In this PR, I was only concerned about the consistency of Github links to the issues.

**Review by ghost — APPROVED** [review]

**Comment by varkor** [comment]
I'll review this soon. However, I feel slightly uncomfortable, because @punitkoura had asked to take this issue. It would be a shame to waste this effort, though.

If @punitkoura is happy with it, perhaps we could go with this pull request, but  @punitkoura can be added as a [co-author](https://help.github.com/en/articles/creating-a-commit-with-multiple-authors) for doing the initial work? Then I can direct @punitkoura to another easy starter issue.

**Comment by punitkoura** [comment]
Hi @varkor, these changes look good to me. I was also working on the same, but then saw this PR and therefore didn't raise another PR. It'd be really nice if I could be added as a co-author though :)

File: src/librustc/session/mod.rs [hunk 1]
```diff
@@ -1317,7 +1317,7 @@ fn validate_commandline_args_with_session_available(sess: &Session) {
        sess.opts.prints.iter().all(|&p| p == PrintRequest::NativeStaticLibs) {
         sess.err("Profile-guided optimization does not yet work in conjunction \
```
Comment (varkor) [inline src/librustc/session/mod.rs]:
```suggestion
        sess.err("profile-guided optimization does not yet work in conjunction \
```
Let's make this quick drive-by fix while we're at it.

File: src/librustdoc/core.rs [hunk 2]
```diff
@@ -402,7 +402,8 @@ pub fn run_core(options: RustdocOptions) -> (clean::Crate, RenderInfo, RenderOpt
             fn report_deprecated_attr(name: &str, diag: &errors::Handler) {
                 let mut msg = diag.struct_warn(&format!("the `#![doc({})]` attribute is \
                                                          considered deprecated", name));
-                msg.warn("please see https://github.com/rust-lang/rust/issues/44136");
+                msg.warn("please see issue \
```
Comment (varkor) [inline src/librustdoc/core.rs]:
```suggestion
                msg.warn("see issue \
```

File: src/librustdoc/config.rs [hunk 3]
```diff
@@ -546,7 +546,7 @@ fn check_deprecated_options(matches: &getopts::Matches, diag: &errors::Handler)
         if matches.opt_present(flag) {
             let mut err = diag.struct_warn(&format!("the '{}' flag is considered deprecated",
                                                     flag));
-            err.warn("please see https://github.com/rust-lang/rust/issues/44136");
+            err.warn("please see issue #44136 <https://github.com/rust-lang/rust/issues/44136>");
```
Comment (varkor) [inline src/librustdoc/config.rs]:
```suggestion
            err.warn("see issue #44136 <https://github.com/rust-lang/rust/issues/44136>");
```

**Comment by varkor** [comment]
It'd be nice to normalise the following strings that appear:
- "for more context, see [...]"
- "for more information, see [...]"
- "See [...] for details"
- "see [...] for more details"
- "see [...] for more information"

I personally prefer "see [...] for more information", as it has the link early on.

Apart from that, this looks good. If you could squash the `--bless`/rebase/tidy commits and add @punitkoura as a co-author, I'll approve it.

**Comment by eupn** [comment]
> It'd be nice to normalise the following strings that appear:
> 
>     * "for more context, see [...]"
> 
>     * "for more information, see [...]"
> 
>     * "See [...] for details"
> 
>     * "see [...] for more details"
> 
>     * "see [...] for more information"
> 
> 
> I personally prefer "see [...] for more information", as it has the link early on.
> 
> Apart from that, this looks good. If you could squash the `--bless`/rebase/tidy commits and add @punitkoura as a co-author, I'll approve it.

@punitkoura don't you mind if I'll use your email `pkoura@andrew.cmu.edu` as `Co-authored-by` for commits?

**Comment by punitkoura** [comment]
@eupn, yes, that email id works
Name - Punit Singh Koura
Email - pkoura@andrew.cmu.edu

**Comment by eupn** [comment]
@varkor

**Comment by varkor** [comment]
Thanks!

@bors r+

**Comment by Centril** [comment]
Failed in https://github.com/rust-lang/rust/pull/63176#issuecomment-517075965, @bors r-

Please `--bless --compare-mode=nll` to fix the issues.

**Comment by eupn** [comment]
@Centril

**Comment by Mark-Simulacrum** [comment]
@bors p=1

since this is pretty conflict-prone

**Comment by varkor** [comment]
@eupn: there have been more merge conflicts, so you'll need to rebase and force push again.

**Comment by ProgrammaticNajel** [comment]
Ping from triage. @eupn any updates on this? Thanks.

**Comment by eupn** [comment]
I’ll rebase it soon!

**Comment by JohnTitor** [comment]
Ping from triage: @eupn any updates on this?

**Comment by JohnTitor** [comment]
Ping from triage: @eupn any updates on this?

**Comment by joelpalmer** [comment]
Ping from Triage: closing due to inactivity. Thanks for the PR @eupn! If you do rebase and have updates, please re-open.

**Pull request closed without merging**